### PR TITLE
Cwagent 1.24

### PR DESCRIPTION
**Pull request by ben851**
## What happens when your PR merges?
Staging cloud watch agent will be updated with a new configuration in preparation for the migration to K8s V1.24

* Tested in my scratch environment against K8s v1.23 and v1.24

## What are you changing?
- [X] Changing kubernetes configuration

## Provide some background on the changes
This configuration change tells the cwagent and fluentd to pull logs from containerd in addition to docker and send them to cloudwatch.

## Checklist if releasing new version:
- [ ] I made sure that the changes are as expected in [Notify staging](https://staging.notification.cdssandbox.xyz/)

## Checklist if making changes to Kubernetes:
- [X] I know how to get kubectl credentials in case it catches on fire

## After merging this PR
- [ ] I have verified that the tests / deployment actions succeeded
- [ ] I have verified that any affected pods were restarted successfully
- [ ] I have verified that I can still log into [Notify production](https://notification.canada.ca)
- [ ] I have verified that the smoke tests still pass on production
- [ ] I have communicated the release in the #notify Slack channel.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,6 +77,11 @@ encrypt-staging:
 	zip .env.zip .env &&\
 	aws kms encrypt --key-id a92df413-fc30-4f3e-8047-7433e1a8ad02 --plaintext fileb://.env.zip --output text --query CiphertextBlob --region ca-central-1 | base64 --decode > .env.zip.enc.aws
 
+encrypt-staging:
```
Comment (jimleroyer) [inline Makefile:80]:
```suggestion
encrypt-scratch:
```

File: env/scratch/cwagent-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+kind: ConfigMap
+metadata:
+  name: prometheus-cwagentconfig
+  namespace: amazon-cloudwatch
+apiVersion: v1
+data:
+  # cwagent json config
+  cwagentconfig.json: |
```
Comment (jimleroyer) [inline env/scratch/cwagent-configmap.yaml:8]: So this is the new CloudWatch agent to use for newer AWS kubernetes setup?

File: env/scratch/cwagent-fluentd-quickstart.yaml [hunk 3]
```diff
@@ -0,0 +1,654 @@
...
+            path: /var/log
+        - name: varlibdockercontainers
+          hostPath:
+            path: /var/lib/docker/containers
+        - name: runlogjournal
+          hostPath:
+            path: /run/log/journal
+        - name: dmesg
+          hostPath:
+            path: /var/log/dmesg
```
Comment (jimleroyer) [inline env/scratch/cwagent-fluentd-quickstart.yaml:654]: How was this big file generated?

File: env/scratch/cwagent-configmap.yaml (hunk 2 above)
Comment (ben851) [reply env/scratch/cwagent-configmap.yaml:8]: This one and the none in the staging folder yep

File: env/scratch/cwagent-fluentd-quickstart.yaml (hunk 3 above)
Comment (ben851) [reply env/scratch/cwagent-fluentd-quickstart.yaml:654]: Mostly a copy/paste from AWS's quickstart repo (which is where it came from originally judging by the file name)
[Cloudwatch Quickstart](https://github.com/aws-samples/amazon-cloudwatch-container-insights/tree/main/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/quickstart)

I then did a diff between the two and added back the stuff that was relevant to Notify.

File: env/scratch/cwagent-configmap.yaml (hunk 2 above)
Comment (jimleroyer) [reply env/scratch/cwagent-configmap.yaml:8]: So no production as we test things first. Not a bad idea hehe.

**Review by jimleroyer — APPROVED** [review]

**Pull request merged**